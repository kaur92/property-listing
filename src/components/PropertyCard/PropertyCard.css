.card {
  width: 80%;
  height: 250px;
  margin: 20px auto;
  display: flex;
  border-radius: 5px;
  padding: 5px;
  box-shadow: 0px 0px 2px 0px rgba(200, 200, 200, 1);
}

.card:hover {
  background: #f5f5f5;
}

.card:hover a {
  text-decoration: underline #000;
}

.card a {
  text-decoration: none;
}

.card-image {
  width: 50%;
  display: inline-block;
  position: relative;
  height: 250px;
  border-radius: 5px;
}

.card-image img {
  width: 100%;
  height: 100%;
}

.card-image .status {
  position: absolute;
  top: 5px;
  left: 5px;
  color: white;
  font-weight: bold;
  padding: 5px;
}

.status-active {
  background: green;
}

.status-inactive {
  background: black;
}

.card-details {
  display: inline-block;
  padding: 10px;
}

.price {
  padding-bottom: 10px;
}

.price label {
  font-size: 10px;
}

.price span {
  display: block;
  font-size: 24px;
  font-weight: 700;
}

.address {
  margin-top: 20px;
  font-weight: 600;
  font-size: 20px;
}

@media only screen and (max-width: 700px) {
  .card {
    display: block;
    height: auto;
  }

  .card-image, .card-details {
    width: 100%;
  }
}
