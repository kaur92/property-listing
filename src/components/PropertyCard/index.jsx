import React from "react";
import { numberWithCommas } from "../../utils/StringFormatter";
import "./PropertyCard.css";

const PropertyCard = ({ property }) => {
  const {
    image,
    status,
    askingPrice,
    numberOfBedrooms,
    propertyType,
    address,
  } = property;

  return (
    <div>
      <div className="card">
        <div className="card-image">
          <img src={image} alt=""></img>
          {status && <div className="status status-active">Active</div>}
          {!status && <div className="status status-inactive">Inactive</div>}
        </div>
        <div className="card-details">
          <div className="price">
            <label>Asking Price</label>
            <span>Â£{numberWithCommas(askingPrice)}</span>
          </div>
          <div className="bathroom">
            <label>{`${numberOfBedrooms} bed ${propertyType} for sale`}</label>
          </div>
          <div className="address">
            <p>{address}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default PropertyCard;
