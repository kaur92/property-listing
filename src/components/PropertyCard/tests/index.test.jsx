import renderer from "react-test-renderer";
import PropertyCard from "../index";

const testProperty = {
  id: "bb32dc153a844c01a051ab48fa543c6f",
  address: "Gloucester Road, London SW7 4RT",
  image:
    "https://lid.zoocdn.com/u/1024/768/2b3d21c22e792b3093479bed79b84b931e4cc062.jpg:p",
  askingPrice: 50000,
  status: true,
  numberOfBedrooms: 1,
  propertyType: "flat",
};

describe("<PropertyCard /> tests", () => {
  it("renders correctly", () => {
    const cardComponent = renderer
      .create(<PropertyCard property={testProperty} />)
      .toJSON();
    expect(cardComponent).toMatchSnapshot();
  });
});
