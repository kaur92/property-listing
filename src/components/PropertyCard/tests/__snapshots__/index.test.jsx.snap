// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PropertyCard /> tests renders correctly 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        alt=""
        src="https://lid.zoocdn.com/u/1024/768/2b3d21c22e792b3093479bed79b84b931e4cc062.jpg:p"
      />
      <div
        className="status status-active"
      >
        Active
      </div>
    </div>
    <div
      className="card-details"
    >
      <div
        className="price"
      >
        <label>
          Asking Price
        </label>
        <span>
          Â£
          50,000
        </span>
      </div>
      <div
        className="bathroom"
      >
        <label>
          1 bed flat for sale
        </label>
      </div>
      <div
        className="address"
      >
        <p>
          Gloucester Road, London SW7 4RT
        </p>
      </div>
    </div>
  </div>
</div>
`;
